tests/os/SpawnTest.php

```diff
@@ -0,0 +1,104 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+use type Facebook\HackTest\HackTest;
+use function HH\Lib\Experimental\OS\posix_spawnp;
+use type HH\Lib\Experimental\OS\posix_spawn_file_actions_adddup2;
+use namespace HH\Lib\IO;
+use function Facebook\FBExpect\expect;
+use namespace \HH\Lib\_Private\_OS;
+
+final class SpawnTest extends HackTest {
+  public async function testBasicUsage(): Awaitable<void> {
+    list($read_pipe, $write_pipe) = IO\pipe();
+
+    using $read_pipe->closeWhenDisposed();
+    try {
+      $pid = posix_spawnp(
+        "ls",
+        vec[
+          new posix_spawn_file_actions_adddup2(
+            $write_pipe->getFileDescriptor(),
+            _OS\STDOUT_FILENO,
+          ),
+        ],
+        shape(),
+        vec["ls", "src/os"],
```

Comment:
use `__DIR__` (file directory) to make this independent of the current working directory